app/models/user_notification.rb

```diff
@@ -25,4 +25,14 @@ class UserNotification < ActiveRecord::Base
   # Associations
   belongs_to :user
   belongs_to :notification
+
+  private def generate_token
+    loop do
+      self[token] = SecureRandom.urlsafe_base64
```

Comment:
What's `self[token]` for?

Reply:
I guess the logic is very similar to user_password_verification line 70 in generate_token(column)